### PR TITLE
STU-230 | Block: Export to s3

### PR DESCRIPTION
**Pull request by LatentDream**

### CODE REVIEW

File: blocks/DATA/EXPORT/EXPORT_S3/EXPORT_S3.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    bucket : str
+        The S3 bucket to upload the file to.
+    region : str
+        The AWS region (default is "us-east-1").
+    enable_overwrite : bool, optional
+        Whether to overwrite the file if it already exists (default is False).
+    file : File
+        The file to be uploaded to the S3 bucket.
+    objet_name: Optional[String]
+        Flojoy input to dynamically provide a string for the name of the object in S3. If not specified, the name of the file will be used.
```
[the remark this answers is not in the document]
Comment (dstrande) [reply blocks/DATA/EXPORT/EXPORT_S3/EXPORT_S3.py:80]: Typo

Also I try to keep the order in the docstring the same as the order in the func args.

[the remark this answers is not in the document]
Comment (LatentDream) [reply blocks/DATA/EXPORT/EXPORT_S3/EXPORT_S3.py:80]: The fact that you can't test it easily means the doc isn't insane.
- Either I put a description in the PR and I do a Flojoy article describing the process article (not only on that, but with this block and some hardwares)
- Or I update the doc to explain how to set up an S3 bucket.

[the remark this answers is not in the document]
Comment (LatentDream) [reply blocks/DATA/EXPORT/EXPORT_S3/EXPORT_S3.py:80]: Thanks for the typos correction, and okok! I'll regenerate the docs with your change :) Thank you

**Comment by LatentDream** [comment]
@dstrande, here is the process if you want to test it:
# Create a S3 bucket

# A general one, basic option

# Create a `IAM` user

# Attach a policy

# Create one with the minimum permission
**Don't forget to change the bucket name for yours!**

Here is the json:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "ListObjectsInBucket",
            "Effect": "Allow",
            "Action": ["s3:ListBucket"],
            "Resource": ["arn:aws:s3:::REPLACE_ME"]
        },
        {
            "Sid": "AllObjectActions",
            "Effect": "Allow",
            "Action": "s3:*Object",
            "Resource": ["arn:aws:s3:::REPLACE_ME/*"]
        }
    ]
}
```
# Once the IAM user created, click on it, and create a access key

# Select local code for the test


# Once created, copy them to the flojoy env vars


# Run the default app, and change the bucket name for yours


# The generated file will be uploaded to the bucket

**Comment by LatentDream** [comment]
If you put the `s3:ListAllMyBuckets` permission, the block will display the available buckets if the bucket doesn't exist.

**Review by dstrande — APPROVED** [review]
lgtm

**Pull request merged**